### PR TITLE
[trainer,data] fix: Support merging `extra_info` for `main_ppo_sync` & update TQ dependency

### PR DESCRIPTION
**Pull request by 0oshowero0**
### What does this PR do?

- Fix `BatchMeta.concat` failure on the TransferQueue path. #6270 added per-rank metrics (e.g. `perf/max_memory_allocated_gb`) to `extra_info`, breaking the assumption that values under the same key in `extra_info` are identical across ranks during concatenation.
- Make TransferQueue a default dependency of verl
- Bump TransferQueue version to v0.1.7 (https://github.com/Ascend/TransferQueue/releases/tag/v0.1.7)

CC @liziniu

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -24,3 +24,4 @@ fastapi
 latex2sympy2_extended
 math_verify
 tensorboard
+TransferQueue==0.1.7
```
Comment (wuxibin89) [inline requirements.txt:27]: Clean up TransferQueue in `setup.py`

File: verl/trainer/main_ppo_sync.py [hunk 2]
```diff
@@ -1771,17 +1694,20 @@ def run(self, config):
...
+
+            trainer = PPOTrainer(
+                config=config,
+                role_worker_mapping=self.role_worker_mapping,
+                resource_pool_manager=self.resource_pool_manager,
+            )
+            trainer.init_workers()
+            trainer.fit()
+        finally:
+            tq.close()
```
Comment (wuxibin89) [inline verl/trainer/main_ppo_sync.py:1765]: Please also help to clean up the debug print in this file.

**Comment by liziniu** [comment]
The functionality looks good to me. 

A few notes from my review:

1. The approach of extracting `extra_info` before `BatchMeta.concat()` and merging manually effectively bypasses the `raise ValueError` in TransferQueue's `BatchMeta.concat` for conflicting extra_info values. This is the correct fix point since `BatchData.concat()` is the sole merge entry in the sync trainer's dispatch/collect path.

2. Minor suggestion (from Claude): consider using a plain `dict` instead of `defaultdict(list)` to avoid potential side effects when accessing non-existent keys downstream:
   ```python
   merged_extra_info = {}
   for meta in data:
       for k, v in meta.extra_info.items():
           merged_extra_info.setdefault(k, []).append(v)
       meta.extra_info = {}
   ```

LGTM overall.

**Review by wuxibin89 — APPROVED** [review]
